cli/options.go

```diff
@@ -365,6 +370,11 @@ func WithExtension(name string, typ any) ProjectOptionsFn {
 	}
 }
 
+// Append listener to event
+func (o *ProjectOptions) WithListener(listeners []loader.Listener) {
+	o.Listeners = listeners
```

Comment:
nit: Better make this a appending function so it can be used multiple times
```suggestion
func (o *ProjectOptions) WithListener(listener loader.Listener) {
	o.Listeners = append(o.Listeners, listener)
```

Reply:
https://github.com/compose-spec/compose-go/pull/571/commits/6e4fb2bfbf56a2bd449bd627c3d3469fd331da2c#diff-0953236283edf016dbef569cb026a05dba9df5e7934af11b82499e3bec8fe006R374